src/main/webapp/WEB-INF/queue.xml

```diff
@@ -102,7 +102,10 @@
         <rate>5/s</rate>
         <bucket-size>20</bucket-size>
         <retry-parameters>
-            <task-retry-limit>2</task-retry-limit>
+            <task-retry-limit>5</task-retry-limit>
+            <min-backoff-seconds>2</min-backoff-seconds>
+            <max-backoff-seconds>4</max-backoff-seconds>
+            <max-doublings>0</max-doublings>
```

Comment:
Try to tweak these values so that we don't exhaust all retries within a few seconds. I don't mind increasing retry gap after each failure if that is possible.

Reply:
@xpdavid can we know how long it takes to complete indexing/ or for entities to be visible, so that then we can accordingly set rhe `min-backoff-seconds` so that by second retry itself it should pass as we will give enough delay between retry.


> I don't mind increasing retry gap after each failure if that is possible.

Its possible. The gap increases linearly right now, however I can make some changes to increase it as expected. 

https://cloud.google.com/appengine/docs/standard/java/taskqueue/push/retrying-tasks

Reply:
Should be done within 1 second. 

BTW, do we really need such a big retry limit? For a task to fail in a request, we need 7s. Therefore, I think modifying  `InstructorCourseJoinEmailWorkerAction.java` and `StudentCourseJoinEmailWorkerAction` should be enough. If a task won't success after 14s (7s * 2), then I think it is not GAE's problem but our problem.

Reply:
I don't mind retrying over a longer period, even one minute, as long as we keep increasing the gap and don't spam the server. I have the impression that occasionally the persistence can take an unusual long time.